src/services/signatureHelp.ts

```diff
@@ -29,6 +29,13 @@ namespace ts.SignatureHelp {
             return undefined;
         }
 
+        if (shouldCarefullyCheckContext(triggerReason)) {
+            // In the middle of a string, don't provide signature help unless the user explicitly requested it.
+            if (isInString(sourceFile, position, startingToken)) {

```

Comment:
we still need to do some additional checks here.

Reply:
e.g.:
(: in a parenthesized expression
,: not in an object literal, array literal, or comma expression
default: instring | incomment | injsxtext,